### PR TITLE
feat: add va spec pydantic models

### LINKED ISSUE #3: create pydantic models
Create Pydantic models using the [1.x branch](https://github.com/ga4gh/va-spec/tree/1.x)

### PR DESCRIPTION
**Pull request by korikuzma**
close #3

* VA Spec (and upstream specs) are still changing. Need to create a snapshot for now. I believe the models will be in a stable spot by the end of the year. 
* I realize I did this on the wrong branch name.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -133,6 +137,7 @@ ignore = [
 # S101 - assert
 # B011 - assert-false
 "tests/*" = ["ANN001", "ANN2", "ANN102", "S101", "B011"]
+"src/ga4gh/cat_vrs/*models.py" = ["ANN102"]
```
Comment (jsstevenson) [inline pyproject.toml:140]: intentional? is this a submodules thing?

Comment (korikuzma) [reply pyproject.toml:140]: Ope thank you. Copy/paste error

File: src/ga4gh/va_spec/profiles/assay_var_effect.py [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+class AssayVariantEffectFunctionalClassificationStatement(StatementBase):
+    """A statement that assigns a functional classification to a variant effect from a functional assay."""
+
+    model_config = ConfigDict(use_enum_values=True)
+
+    type: Literal["AssayVariantEffectFunctionalClassificationStatement"] = Field(
+        "AssayVariantEffectFunctionalClassificationStatement",
+        description="MUST be 'AssayVariantEffectFunctionalClassificationStatement'.",
+    )
+    subjectVariant: MolecularVariation | CategoricalVariant | IRI = Field(  # noqa: N815
```
[the remark this answers is not in the document]
Comment (korikuzma) [reply src/ga4gh/va_spec/profiles/assay_var_effect.py:51]: I was going back and forth on this. For fields, it is annoying. But assuming we add more validator code in the future, we'd want to ensure we're using correct case. I have no strong feelings, so if you'd like I can ignore.

[the remark this answers is not in the document]
Comment (jsstevenson) [reply src/ga4gh/va_spec/profiles/assay_var_effect.py:51]: I don't feel strongly about it either

**Review by jsstevenson — APPROVED** [review]

File: src/ga4gh/va_spec/profiles/assay_var_effect.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (korikuzma) [reply src/ga4gh/va_spec/profiles/assay_var_effect.py:51]: I'm going to add it to the config. We have 43 instances

**Review by jsstevenson — APPROVED** [review]

**Pull request merged**